src/pages/Service.js

```diff
@@ -20,6 +21,16 @@ class Service extends Component {
 
 
   componentDidMount() {
+    if (process.env.NODE_ENV !== 'production') {
+      // Allow querystrings to set data, which is used in joplin for livepreview
+      const query = parse(this.props.location.search);
```

Comment:
Live preview currently seems very coupled to the service page. Any plans for how we might achieve preview for all pages?
Also, would postMessage work for passing content to the iframe in a cleaner/safer way?
https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage

Reply:
It's definitely coupled. I'm honestly not sure the best way to get this working for all pages. Googling around shows some examples of listening to react router and then setting component props. I'm open to any suggestions on how this should fit in with the app.

I considered postmessage for this but I wasn't sure where a good place to put a listener would be. Since the child doesn't (currently) need to pass data to the parent, I decided to go the react router way. postmessage might make things easier regarding the "how does this work for all pages" question. I think either way is safe since it won't be active on a resident-facing site.

TL;DR I am terrible at javascript.